lib/ClangImporter/ImporterImpl.h

```diff
@@ -1891,43 +1891,20 @@ static inline Type applyToFunctionType(
   return type;
 }
 
-inline Optional<const clang::EnumDecl *> findAnonymousEnumForTypedef(
-    const ASTContext &ctx,
-    const clang::TypedefType *typedefType) {
-  auto *typedefDecl = typedefType->getDecl();
-  auto *lookupTable = ctx.getClangModuleLoader()->findLookupTable(typedefDecl->getOwningModule());
-
-  auto foundDecls = lookupTable->lookup(
-      SerializedSwiftName(typedefDecl->getName()), EffectiveClangContext());
-
-  auto found = llvm::find_if(foundDecls, [](SwiftLookupTable::SingleEntry decl) {
-    return decl.is<clang::NamedDecl *>() &&
-        isa<clang::EnumDecl>(decl.get<clang::NamedDecl *>());
-  });
-
-  if (found != foundDecls.end())
-    return cast<clang::EnumDecl>(found->get<clang::NamedDecl *>());
-
-  // If a swift_private attribute has been attached to the enum, its name will
-  // be prefixed with two underscores
-  llvm::SmallString<32> swiftPrivateName;
-  swiftPrivateName += "__";
-  swiftPrivateName += typedefDecl->getName();
-  foundDecls = lookupTable->lookup(
-      SerializedSwiftName(ctx.getIdentifier(swiftPrivateName)),
-      EffectiveClangContext());
-
-  auto swiftPrivateFound =
-      llvm::find_if(foundDecls, [](SwiftLookupTable::SingleEntry decl) {
-        return decl.is<clang::NamedDecl *>() &&
-               isa<clang::EnumDecl>(decl.get<clang::NamedDecl *>()) &&
-               decl.get<clang::NamedDecl *>()
-                   ->hasAttr<clang::SwiftPrivateAttr>();
-      });
-
-  if (swiftPrivateFound != foundDecls.end())
-    return cast<clang::EnumDecl>(swiftPrivateFound->get<clang::NamedDecl *>());
-
+inline Optional<const clang::EnumDecl *>
+findAnonymousEnumForTypedef(const ASTContext &ctx,
+                            const clang::TypedefType *typedefType) {
+
+  assert(typedefType && "Must provide non-null typedef");
+  const clang::TypedefNameDecl *typedefDecl = typedefType->getDecl();
+  const clang::DeclContext *declContext = typedefDecl->getDeclContext();
+  for (const clang::Decl *siblingDecl : declContext->decls()) {
```

Comment:
I'm also a little worried about performance here. I think this is slower than a lookup table.

Reply:
I would also guess that it is slower, but the lookup table doesn't really work once NS_SWIFT_NAME gets involved. I briefly attempted essentially a second lookup table, but one that makes Clang pointer -> Clang node pointer. Trouble is I wasn't sure where to store the mapping, the lookup tables get serialized into the PCM, then read back out, which seems like an awful headache to replicate. Otherwise I'm not sure how to find it aside from walking the AST.